src/pages/workspaces/workspace/analysis/modals/export-analysis-modal/ExportAnalysisModal.test.ts

```diff
@@ -0,0 +1,362 @@
+import { render, screen, within } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { h } from 'react-hyperscript-helpers'
+import { WorkspaceInfo, WorkspaceWrapper } from 'src/libs/workspace-utils'
+import {
+  useAnalysesExportState
+} from 'src/pages/workspaces/workspace/analysis/modals/export-analysis-modal/export-analysis-modal.state'
+import { toolLabels } from 'src/pages/workspaces/workspace/analysis/utils/tool-utils'
+import { asMockedFn } from 'src/testing/test-utils'
+
+import { ExportAnalysisModal } from './ExportAnalysisModal'
+
+
+type ModalExports = typeof import('src/components/Modal')
+jest.mock('src/components/Modal', (): ModalExports => {
+  const modalMock = jest.requireActual('src/components/Modal.mock')
+  return modalMock.mockModalModule()
+})
+
+type ExportAnalysisModalStateExports = typeof import('./export-analysis-modal.state')
+jest.mock('./export-analysis-modal.state', (): ExportAnalysisModalStateExports => ({
+  ...jest.requireActual('./export-analysis-modal.state'),
+  useAnalysesExportState: jest.fn()
+}))
+
+type UtilsExports = typeof import('src/libs/utils')
+type LodashFpExports = typeof import('lodash/fp')
+jest.mock('src/libs/utils', (): UtilsExports => {
+  const _ = jest.requireActual<LodashFpExports>('lodash/fp')
+  return ({
+    ...jest.requireActual('src/libs/utils'),
+    isValidWsExportTarget: jest.fn().mockImplementation(_.curry((sourceWs: WorkspaceWrapper, destWs: WorkspaceWrapper) => {
+      // mock this to have a much simpler check then the real implementation
+      return (sourceWs.workspace.workspaceId !== destWs.workspace.workspaceId)
+    }))
+  })
+})
+
+// This is for the AutoSizer component. It requires screen dimensions in order to be tested properly.
+Object.defineProperties(window.HTMLElement.prototype, {
```

Comment:
This feels like it should be made into a reusable testing utility that lives near the AutoSizer...

Reply:
Yes, that's probably true.

I think there's a cleaner approach to getting this done, but we'd have to mock deeper dependencies.

I added this as a function to test-utils. Maybe this is a better approach for now?